rust/datafusion/src/execution/physical_plan/datasource.rs

```diff
@@ -61,6 +73,12 @@ pub struct DatasourcePartition {
     batch_iter: Arc<Mutex<dyn RecordBatchReader + Send + Sync>>,
 }
 
+impl Debug for DatasourcePartition {
```

Comment:
`RecordBatchReader` does not implement `Debug` so had to implement directly here.